rfd/0000-sre-challenge.md

```diff
@@ -0,0 +1,213 @@
+### Design Document for Kubernetes Deployment Scaler
+
+---
+authors: Derek Norrbom (dereknorrbom@gmail.com)
+state: draft
+---
+
+## What
+
+This document outlines the proposed design and implementation plan for a Kubernetes Deployment Scaler API. The API will provide functionality to get and set the replica count of a Kubernetes deployment, utilizing mTLS for secure communication and caching for efficient data retrieval.
+
+## Why
+
+The purpose of this project is to implement a scalable, secure API for managing Kubernetes deployments. This project will demonstrate proficiency in Go, Kubernetes, Docker, and security best practices (mTLS). It will also highlight the ability to design, build, and document an application that can be easily extended and maintained.
+
+## How
+
+### API Structure
+
+The API will provide the following endpoints:
+
+- **GET /healthz**: Health check endpoint that returns "OK" if the server is running.
+- **GET /replica-count**: Returns the current replica count of the specified deployment.
+- **POST /set-replica-count**: Sets the replica count of the specified deployment.
+- **GET /deployments**: Lists all deployments in the specified namespace.
```

Comment:
How will the API support specifying the namespace?

Reply:
I planned to hardcode the namespace value based on the challenge documentation's suggestion to "_Use hardcoded values as much as possible and simply add TODO items showing your thinking._"  To support specifying the namespace, I can modify the API design to accept it as a query parameter.